### PR TITLE
[RIP] Remove legacy GC

### LINKED ISSUE #1912: Test test_gc:testBasicGCWithEmptyInvIdx fails with "GC_POLICY LEGACY timeout 100000"
```
# make pytest TEST=test_gc:testBasicGCWithEmptyInvIdx MOD_ARGS='GC_POLICY LEGACY timeout 100000'
test_gc:testBasicGCWithEmptyInvIdx
Redis master command: redis-server --port 6379 --loadmodule /w/rafi_1/RediSearch/build/redisearch.so GC_POLICY LEGACY timeout 100000 --dir ./logs --dbfilename 00ceb74cdc75405ca2fddfbfb34a84ce.master-1-test_gc_testBasicGCWithEmptyInvIdx-oss.rdb
36243:C 17 Mar 2021 14:23:40.754 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
36243:C 17 Mar 2021 14:23:40.754 # Redis version=6.2.1, bits=64, commit=00000000, modified=0, pid=36243, just started
36243:C 17 Mar 2021 14:23:40.754 # Configuration loaded
36243:M 17 Mar 2021 14:23:40.756 * Increased maximum number of open files to 10032 (it was originally set to 1024).
36243:M 17 Mar 2021 14:23:40.756 * monotonic clock: POSIX clock_gettime
                _._                                                  
           _.-``__ ''-._                                             
      _.-``    `.  `_.  ''-._           Redis 6.2.1 (00000000/0) 64 bit
  .-`` .-```.  ```\/    _.,_ ''-._                                   
 (    '      ,       .-`  | `,    )     Running in standalone mode
 |`-._`-...-` __...-.``-._|'` _.-'|     Port: 6379
 |    `-._   `._    /     _.-'    |     PID: 36243
  `-._    `-._  `-./  _.-'    _.-'                                   
 |`-._`-._    `-.__.-'    _.-'_.-'|                                  
 |    `-._`-._        _.-'_.-'    |           http://redis.io        
  `-._    `-._`-.__.-'_.-'    _.-'                                   
 |`-._`-._    `-.__.-'    _.-'_.-'|                                  
 |    `-._`-._        _.-'_.-'    |                                  
  `-._    `-._`-.__.-'_.-'    _.-'                                   
      `-._    `-.__.-'    _.-'                                       
          `-._        _.-'                                           
              `-.__.-'                                               
36243:M 17 Mar 2021 14:23:40.756 # WARNING: The TCP backlog setting of 511 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.
36243:M 17 Mar 2021 14:23:40.756 # Server initialized
36243:M 17 Mar 2021 14:23:40.756 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
36243:M 17 Mar 2021 14:23:40.758 * <search> Redis version found by RedisSearch : 6.2.1 - oss
36243:M 17 Mar 2021 14:23:40.758 * <search> RediSearch version 99.99.99 (Git=v1.99.5-163-g95808d7a)
36243:M 17 Mar 2021 14:23:40.758 * <search> Low level api version 1 initialized successfully
36243:M 17 Mar 2021 14:23:40.758 * <search> concurrent writes: OFF, gc: ON, prefix min length: 2, prefix max expansions: 200, query timeout (ms): 100000, timeout policy: return, cursor read size: 1000, cursor max idle (ms): 300000, max doctable size: 1000000, max number of search results:  1000000, search pool size: 20, index pool size: 8, 
36243:M 17 Mar 2021 14:23:40.758 * <search> Initialized thread pool!
36243:M 17 Mar 2021 14:23:40.758 * Module 'search' loaded from /w/rafi_1/RediSearch/build/redisearch.so
36243:M 17 Mar 2021 14:23:40.760 * Ready to accept connections
36243:M 17 Mar 2021 14:23:40.853 * <search> Successfully changed configuration for `FORK_GC_CLEAN_THRESHOLD`
        ✅  (OK):        'OK' == 'OK'    test_gc.py:44
        ✅  (OK):        'OK' == 'OK'    test_gc.py:45
        ✅  (OK):        'OK' == 'OK'    test_gc.py:48
        ✅  (OK):        [1L] == [1]     test_gc.py:50
        ✅  (OK):        1L == 1 test_gc.py:52
        ❌  (FAIL):      False == True   test_gc.py:57
        ❌  (FAIL):      [] should contain 'Can not find the inverted index'     test_gc.py:57
        [FAIL]
36243:M 17 Mar 2021 14:23:40.911 * DB saved on disk
36243:signal-handler (1615983820) Received SIGTERM scheduling shutdown...
36243:M 17 Mar 2021 14:23:40.961 # User requested shutdown...
36243:M 17 Mar 2021 14:23:40.961 * Saving the final RDB snapshot before exiting.
36243:M 17 Mar 2021 14:23:40.999 * DB saved on disk
36243:M 17 Mar 2021 14:23:40.999 # Redis is now ready to exit, bye bye...
Test Took: 0 sec
Total Tests Run: 1, Total Tests Failed: 2, Total Tests Passed: -1
Failed Tests Summary:
        test_gc:testBasicGCWithEmptyInvIdx
                ❌  (FAIL):      False == True   test_gc.py:57
                ❌  (FAIL):      [] should contain 'Can not find the inverted index'     test_gc.py:57
make: *** [Makefile:159: pytest] Error 1
```

### PR DESCRIPTION
**Pull request by ashtul**
closes #1912
MOD-4183

### CODE REVIEW

File: src/trie/trie_type.h [hunk 1]
```diff
@@ -61,9 +61,6 @@ Vector *Trie_Search(Trie *tree, const char *s, size_t len, size_t num, int maxDi
  * Otherwise we return an iterator to all strings within maxDist Levenshtein distance */
 TrieIterator *Trie_Iterate(Trie *t, const char *prefix, size_t len, int maxDist, int prefixMode);
 
-/* Get a random key from the trie, and put the node's score in the score pointer. Returns 0 if the
- * trie is empty and we cannot do that */
-int Trie_RandomKey(Trie *t, char **str, t_len *len, double *score);
```
Comment (MeirShpilraien) [inline src/trie/trie_type.h:66]: I believe this API might be useful. Should we keep it?

File: tests/pytests/test_gc.py [hunk 2]
```diff
@@ -33,27 +33,6 @@ def testBasicGC(env):
     env.assertEqual(env.cmd('ft.debug', 'DUMP_NUMIDX', 'idx', 'id'), [[int(i) for i in range(2, 102)]])
     env.assertEqual(env.cmd('ft.debug', 'DUMP_TAGIDX', 'idx', 't'), [['tag1', [int(i) for i in range(2, 102)]]])
 
-def testBasicGCWithEmptyInvIdx(env):
```
Comment (oshadmi) [inline tests/pytests/test_gc.py:36]: Why remove this test? (relevant for fork GC)

File: src/config.c [hunk 3]
```diff
@@ -425,9 +425,10 @@ CONFIG_SETTER(setGcPolicy) {
   if (!strcasecmp(policy, "DEFAULT") || !strcasecmp(policy, "FORK")) {
     config->gcPolicy = GCPolicy_Fork;
   } else if (!strcasecmp(policy, "LEGACY")) {
-    config->gcPolicy = GCPolicy_Sync;
+    QueryError_SetError(status, QUERY_EPARSEARGS, "Legacy GC policy is no longer supported (since 2.6.0)");
```
Comment (oshadmi) [inline src/config.c:428]: Is this a breaking change, or is it already deprecated?
Seems we also need to update the documentation
https://redis.io/docs/stack/search/configuring/#GC_POLICY

Comment (ashtul) [reply src/config.c:428]: Done

**Review by efeldblum-redis — APPROVED** [review]

**Pull request merged**